tests/platform_tests/cli/test_show_platform_dpu.py

```diff
@@ -0,0 +1,183 @@
+    ip_address_list = []
+
+    output_dpu_status = duthost.show_and_parse('show chassis module status')
+
+    for index in range(len(output_dpu_status)):
+        parse_output = output_dpu_status[index]
+        if 'DPU' in parse_output['name']:
+            if parse_output['oper-status'] != 'Offline':
+                index = (parse_output['name'])[-1]
+                # Api will always return the ip address as it is statically configured now
+                # Check will be added for ip address once dynamic implementation is in place
+                ip_address_list.append(module.get_midplane_ip(platform_api_conn, index))
+
+    check_dpu_ping_status(duthost, ip_address_list)
+
+def test_link_flap(duthosts, enum_rand_one_per_hwsku_hostname):
+    """
+    @summary: Verify `Link flap between NPU and DPU`
+    - Bringing all DPU interfaces down.
+    - Deleting the state table from redis db
+    - Checking show system-health dpu to see no output is there.
+      (Since the db entries are deleted and link down, state table would be empty which gets reflected in the cli)
+    - Brining up the interfaces.
+    - State table gets updated automatically after a minute
+    - Checking the show system-health dpu cli to get all the details
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    count_up = 0
+    count_down = 0
+    dpu_db_before = 0
+    dpu_db_after = 0
+
+    # Checking the state of dpu health before the flap
+    dpu_db_before = count_dpu_modules_in_system_health_cli(duthost)
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+
+        if interface != "eth0" and 'eth' in interface:
+            output_intf_down = duthost.shell("ifconfig %s down"%(interface))["stdout_lines"]
+            time.sleep(1)
+
+    # Deleting the DB table entry after bringing down the interface
+    duthost.shell(
+            "for key in `redis-cli -p 6380 -h 127.0.0.1 -n 13 --scan --pattern \"DPU_STATE|DPU*\" `; "
+            "do redis-cli -p 6380 -h 127.0.0.1 -n 13 del $key ; done")
+
+    # Checking the state of dpu health again after bringing down interface and deleting db entry
+    dpu_db = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
+
+    pytest_assert(dpu_db == 0, "Link is not down'{}'".format(duthost.hostname))
+
+    output_interface_cmd = duthost.show_and_parse('show ip interface')
+    for index in range(len(output_interface_cmd)):
+        parse_output = output_interface_cmd[index]
+        interface = parse_output['interface']
+        status = parse_output['admin/oper']
+
+        if interface != "eth0" and 'eth' in interface:
+            if status == "down/down":
+                count_down += 1
+            output_intf_up = duthost.shell("ifconfig %s up"%(interface))["stdout_lines"]
+            time.sleep(1)
```

Comment:
Is 1 second enough for the port to up? Why not combine this sleep with the wait_until check?

Reply:
We dont need to wait actually, the interface face would be up within a second.